cogs/moderation.py

```diff
@@ -131,6 +131,25 @@ def __init__(self, bot):
             await ctx.author.send(f"Could not send a warning to `{target}`.\n"
                                   "They may have DMs disables, or have blocked me.\n\n"
                                   f"Warning text was:\n{warning}")
+            
+    @command(hidden=True, aliases=['prune, p'])
+    async def purge(self, ctx, count: int):
+        """[p]purge <n> [@user] - purge <n> messages. Ping @user to only delete that user's messages of the last <n> messages. Multiple users also supported."""
+        if ctx.channel.permissions_for(ctx.author).manage_messages:
+            specific_users = [i.id for i in ctx.message.mentions]
+            await ctx.message.delete()
+            if count > 100:
+                await ctx.send(f"You are about to purge {count}. Are you sure you want to purge these many messages? (y/n)")
+                reply = await self.bot.wait_for("message", check=lambda message: (message.content.lower().strip() == 'y' or message.content.lower().strip() == 'n') and message.author == ctx.message.author)
+                if not reply or reply.content.lower().strip() == 'n':
+                    return await ctx.send("Cancelled purge.")
+                else:
+                    await ctx.message.delete()
+            async for message in ctx.message.channel.history(limit=count):
```

Comment:
[this](http://discordpy.readthedocs.io/en/rewrite/api.html#discord.TextChannel.delete_messages) is a thing, doing what you did will ratelimit and is also uglier